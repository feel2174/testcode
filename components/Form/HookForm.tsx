import { useForm } from 'react-hook-form';
import React from 'react';
import { FormErrorMessage, FormLabel, FormControl, Input, Button } from '@chakra-ui/react';

const HookForm = (props: { name?: string }) => {
  const {
    handleSubmit,
    register,
    formState: { errors, isSubmitting },
  } = useForm();

  const onSubmit = (values: any) => {
    return new Promise((resolve) => {
      setTimeout(() => {
        alert(JSON.stringify(values, null, 2));
        resolve(values);
      }, 1000);
    });
  };

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <FormControl style={{ display: 'flex', flexDirection: 'column', gap: '30px', margin: '50px 0px' }} isInvalid={errors.name}>
        <div>
          <FormLabel htmlFor="name">First name</FormLabel>
          <Input
            id="name"
            placeholder="name"
            {...register('name', {
              required: 'This is required',
              minLength: { value: 3, message: 'Minimum length should be 4' },
            })}
          />
        </div>
        <div>
          <FormLabel htmlFor="name">{props.name}</FormLabel>
          <Input
            id="nickname"
            placeholder="nickname"
            {...register('nickname', {
              required: 'This is required',
              minLength: { value: 3, message: 'Minimum length should be 4' },
            })}
          />
        </div>
        <FormErrorMessage>{errors.name && errors.name.message}</FormErrorMessage>
      </FormControl>
      <Button mt={4} colorScheme="teal" isLoading={isSubmitting} type="submit">
        Submit
      </Button>
    </form>
  );
};

export default HookForm;
