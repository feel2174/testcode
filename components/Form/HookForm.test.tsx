import React from 'react';
import { render } from '@testing-library/react';
import HookForm from './HookForm';

describe('HookForm', () => {
  test('renders App component', () => {
    render(<HookForm />);
  });
});

// describe('render', () => {
//   it('renders App component', () => {
//     const { container } = render(<HookForm name="호박너구리" />);
//     expect(container).toHaveTextContent('호박너구리');
//   });
// });

test('matches snapshot', () => {
  const { container } = render(<HookForm name="호박너구리" />);
  expect(container).toMatchSnapshot();
});
