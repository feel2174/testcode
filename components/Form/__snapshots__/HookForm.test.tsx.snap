// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <form>
    <div
      class="chakra-form-control css-0"
      role="group"
      style="display: flex; flex-direction: column; gap: 30px;"
    >
      <div>
        <label
          class="chakra-form__label css-uvho5g"
          for="name"
          id="field-2373566330-2-label"
        >
          First name
        </label>
        <input
          class="chakra-input css-0"
          id="name"
          name="name"
          placeholder="name"
        />
      </div>
      <div>
        <label
          class="chakra-form__label css-uvho5g"
          for="name"
          id="field-2373566330-2-label"
        >
          호박너구리
        </label>
        <input
          class="chakra-input css-0"
          id="nickname"
          name="nickname"
          placeholder="nickname"
        />
      </div>
    </div>
    <button
      class="chakra-button css-q1ts0i"
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
